source/sessions/snapshot-sessions.rst

```diff
@@ -0,0 +1,351 @@
+   in the ``ClientSession`` object.
+
+2. Passing that ``snapshotTimestamp`` in the ``atClusterTime`` field of the ``readConcern`` field
+   for subsequent snapshot read operations (for find/aggregate/distinct commands).
+
+Server Command Responses
+========================
+
+To support snapshot reads the server returns the ``atClusterTime`` in
+cursor object it sends to the driver (for both find/aggregate commands).
+
+.. code:: typescript
+
+    {
+        ok : 1 or 0,
+        ... // the rest of the command reply
+        cursor : {
+            ... // the rest of the cursor reply
+            atClusterTime : <BsonTimestamp>
+        }
+    }
+
+The ``atClusterTime`` MUST be stored in the ``ClientSession`` to later be passed as the
+``atClusterTime`` field of the ``readConcern`` with ``snapshot`` level field  in subsequent read operations.
+
+Server Errors
+=============
+1. The server may reply to read commands with a ``SnapshotTooOld`` error if the client's ``atClusterTime`` value is not available in the server's history.
+2. The server will return ``InvalidOptions`` error if both ``atClusterTime`` and ``afterClusterTime`` options are set to true.
+
+Snapshot read commands
+======================
+
+For snapshot reads the driver MUST first obtain ``atClusterTime`` from cursor response of find/aggregate command,
+by specifying ``readConcern`` with ``snapshot`` level field, and store it as ``snapshotTimestamp`` in 
+``ClientSession`` object.
+
+.. code:: typescript
+
+    {
+        find : <string>, // or other read command
+        ... // the rest of the command parameters
+        readConcern :
+        {
+            level : "snapshot"
+        }
+    }
+
+For subsequent reads from same snapshot driver MUST send the ``snapshotTimestamp`` saved in
+the ``ClientSession`` as the value of the ``atClusterTime`` field of the
+``readConcern`` with ``snapshot`` level field:
+
+.. code:: typescript
+
+    {
+        find : <string>, // or other read command
+        ... // the rest of the command parameters
+        readConcern :
+        {
+            level : "snapshot",
+            afterClusterTime : <BsonTimestamp>
+        }
+    }
+
+Lists of commands that support snapshot reads:
+
+1. find
+2. aggregate
+3. distinct
+
+Snapshot read commands errors
+=============================
+
+Drivers MUST NOT retry errors in SnapshotError category for snapshot reads if ``atClusterTime`` is supplied.
+
+Test Plan
+=========
+
+Note: some tests are only relevant to certain deployments. For the purpose of deciding
+which tests to run assume that any deployment that is version 5.0 or higher and is either a
+replica set or a sharded cluster supports snapshot reads.
+The server ``minSnapshotHistoryWindowInSeconds`` parameter SHOULD be configured to match the test execution time.
+
+1.  | The first read in a snapshot session must not send ``atClusterTime``
```

Comment:
This only needs to be included for prose tests.  You don't have to duplicate what's already specified in YAML.